sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/propagation/AWSXRayPropagator.java

```diff
@@ -0,0 +1,238 @@
+ *
+ * To register the X-Ray propagator together with default propagator:
+ * <pre>
+ * OpenTelemetry.setPropagators(
+ *   DefaultContextPropagators.builder()
+ *     .addHttpTextFormat(new HttpTraceContext())
+ *     .addHttpTextFormat(new AWSXRayPropagator()).build());
+ * </pre>
+ */
+public class AWSXRayPropagator implements HttpTextFormat {
+
+  private static final Logger logger = Logger.getLogger(AWSXRayPropagator.class.getName());
+
+  static final String TRACE_HEADER_KEY = "X-Amzn-Trace-Id";
+  private static final char TRACE_HEADER_DELIMITER = ';';
+  private static final char KV_DELIMITER = '=';
+
+  private static final String TRACE_ID_KEY = "Root";
+  private static final String TRACE_ID_VERSION = "1";
+  private static final char TRACE_ID_DELIMITER = '-';
+  private static final int TRACE_ID_FIRST_PART_LENGTH = 8;
+  private static final int TRACE_ID_SECOND_PART_LENGTH = 24;
+
+  private static final String PARENT_ID_KEY = "Parent";
+  private static final int PARENT_ID_LENGTH = 16;
+
+  private static final String SAMPLED_FLAG_KEY = "Sampled";
+  private static final int SAMPLED_FLAG_LENGTH = 1;
+  private static final char IS_SAMPLED = '1';
+  private static final char NOT_SAMPLED = '0';
+
+  private static final List<String> FIELDS = Collections.singletonList(TRACE_HEADER_KEY);
+
+  @Override
+  public List<String> fields() {
+    return FIELDS;
+  }
+
+  @Override
+  public <C> void inject(Context context, @Nullable C carrier, Setter<C> setter) {
+    Objects.requireNonNull(context, "context");
+    Objects.requireNonNull(setter, "setter");
+
+    Span span = TracingContextUtils.getSpanWithoutDefault(context);
+    if (span == null) {
+      return;
+    }
+
+    SpanContext spanContext = span.getContext();
+
+    String otTraceId = spanContext.getTraceId().toLowerBase16();
+    String xrayTraceId = TRACE_ID_VERSION + TRACE_ID_DELIMITER +
+        otTraceId.substring(0, TRACE_ID_FIRST_PART_LENGTH) + TRACE_ID_DELIMITER +
+        otTraceId.substring(TRACE_ID_FIRST_PART_LENGTH);
+    String parentId = spanContext.getSpanId().toLowerBase16();
+    char samplingFlag = spanContext.getTraceFlags().isSampled() ? IS_SAMPLED : NOT_SAMPLED;
+
+    String traceHeader = TRACE_ID_KEY + KV_DELIMITER + xrayTraceId + TRACE_HEADER_DELIMITER +
+        PARENT_ID_KEY + KV_DELIMITER + parentId + TRACE_HEADER_DELIMITER +
+        SAMPLED_FLAG_KEY + KV_DELIMITER + samplingFlag;
+    setter.set(carrier, TRACE_HEADER_KEY, traceHeader);
+  }
+
+  @Override
+  public <C> Context extract(Context context, C carrier, Getter<C> getter) {
+    checkNotNull(carrier, "carrier");
+    checkNotNull(getter, "getter");
+
+    SpanContext spanContext = getSpanContextFromHeader(carrier, getter);
+    return TracingContextUtils.withSpan(DefaultSpan.create(spanContext), context);
+  }
+
+  @SuppressWarnings("StringSplitter")
+  private static <C> SpanContext getSpanContextFromHeader(C carrier, Getter<C> getter) {
+    try {
```

Comment:
Instead of a large catch-all exception, only catch where an exception could happen. Even if we're worried about missing something, this isn't the right layer to have a catch-all.